ion/integration_test.go

```diff
@@ -0,0 +1,807 @@
+	"timestamp_0000-01-00.ion",
+	"timestamp_0000-01-00T.ion",
+	"timestamp_0000-01-01.ion",
+	"timestamp_0000-01-01T.ion",
+	"timestamp_0000-01T.ion",
+	"timestamp_0000-12-31.ion",
+	"timestamp_0000T.ion",
+	"timestamp_0001-00-00.ion",
+	"timestamp_0001-00-00T.ion",
+	"timestamp_0001-00-01.ion",
+	"timestamp_0001-00-01T.ion",
+	"timestamp_0001-00T.ion",
+	"timestamp_0001-01-00.ion",
+	"timestamp_0001-01-00T.ion",
+	"timestamp_10.ion",
+	"timestamp_11.ion",
+	"timestamp_5.ion",
+	"timestamp_6.ion",
+	"timestamp_7.ion",
+	"timestamp_8.ion",
+	"timestamp_9.ion",
+	"type_3_length_0.10n",
+	"year_3.ion",
+}
+
+var equivsSkipList = []string{
+	"annotatedIvms.ion",
+	"bigInts.ion",
+	"clobs.ion",
+	"localSymbolTableAppend.ion",
+	"localSymbolTableNullSlots.ion",
+	"localSymbolTableWithAnnotations.ion",
+	"localSymbolTables.ion",
+	"localSymbolTablesValuesWithAnnotations.ion",
+	"nonIVMNoOps.ion",
+	"sexps.ion",
+	"stringUtf8.ion",
+	"strings.ion",
+	"structsFieldsDiffOrder.ion",
+	"structsFieldsRepeatedNames.ion",
+	"systemSymbols.ion",
+	"systemSymbolsAsAnnotations.ion",
+	"timestampSuperfluousOffset.10n",
+	"timestamps.ion",
+	"timestampsLargeFractionalPrecision.ion",
+}
+
+var nonEquivsSkipList = []string{
+	"bools.ion",
+	"decimals.ion",
+	"documents.ion",
+	"floats.ion",
+	"floatsVsDecimals.ion",
+	"localSymbolTableWithAnnotations.ion",
+	"structs.ion",
+	"symbolTables.ion",
+	"symbolTablesUnknownText.ion",
+	"symbols.ion",
+	"timestamps.ion",
+}
+
+func TestBinaryRoundTrip(t *testing.T) {
+	readFilesAndTest(t, goodPath, binaryRoundTripSkipList, func(t *testing.T, path string) {
+		testBinaryRoundTrip(t, path)
+	})
+}
+
+func TestTextRoundTrip(t *testing.T) {
+	readFilesAndTest(t, goodPath, textRoundTripSkipList, func(t *testing.T, path string) {
+		testTextRoundTrip(t, path)
+	})
+}
+
+func TestLoadBad(t *testing.T) {
+	readFilesAndTest(t, badPath, malformedIonsSkipList, func(t *testing.T, path string) {
+		testLoadBad(t, path)
+	})
+}
+
+func TestEquivalency(t *testing.T) {
+	readFilesAndTest(t, equivsPath, equivsSkipList, func(t *testing.T, path string) {
+		testEquivalency(t, path, true)
+	})
+}
+
+func TestNonEquivalency(t *testing.T) {
+	readFilesAndTest(t, nonEquivsPath, nonEquivsSkipList, func(t *testing.T, path string) {
+		testEquivalency(t, path, false)
+	})
+}
+
+// Execute round trip testing for BinaryWriter. Create a BinaryWriter, using the BinaryWriter creat a
+// TextWriter and back to BinaryWriter again. Validate that the first and last Writers are equal.
+func testBinaryRoundTrip(t *testing.T, fp string) {
+	fileBytes := loadFile(t, fp)
+
+	// Make a binary writer from the file
+	binWriter, buf := makeBinaryWiter(t, fileBytes)
+
+	// Make a text writer from the binary writer
+	_, str := makeTextWriter(t, buf.String())
+
+	// Make another binary writer using writer
+	binWriter2, _ := makeBinaryWiter(t, []byte(str.String()))
+
+	// Compare the 2 binary writers
+	if !reflect.DeepEqual(binWriter, binWriter2) {
```

Comment:
Are you suggesting to change the comparison between ```buf``` and the buffer for ```binWriter2```?
I tested it and the results are different from current comparison.

Reply:
Per our offline discussion, we want to compare the values that can be read from each encoding rather than comparing the `Writer`s or buffers directly.

Reply:
Done